src/aioquic/h3/connection.py

```diff
@@ -304,6 +316,35 @@ def validate_trailers(headers: Headers) -> None:
         required_pseudo_headers=frozenset(),
     )
 
+class H3Capsule:
+    """
+    Represents the Capsule concept defined in
+    https://ietf-wg-masque.github.io/draft-ietf-masque-h3-datagram/draft-ietf-masque-h3-datagram.html#name-capsules.
+    """
+    def __init__(self, type: int, data: bytes) -> None:
+        self.type = type
+        self.data = data
+
+    @staticmethod
+    def decode(data: bytes) -> any:
+        """
+        Returns an H3Capsule representing the given bytes.
+        """
+        buffer = Buffer(data=data)
+        type = buffer.pull_uint_var()
+        length = buffer.pull_uint_var()
+        return H3Capsule(type, buffer.pull_bytes(length))
+
+    def encode(self) -> bytes:
```

Comment:
I might be missing something, but where is the code that read/writes the encoded capsules to/from the data stream?

Reply:
I was thinking about using this in our WebTransport over HTTP/3 server (https://github.com/web-platform-tests/wpt/blob/master/tools/webtransport/h3/webtransport_h3_server.py).

I think it'd be beneficial to have this here because many WebTransport over HTTP/3 servers will use it.

Reply:
Oh that makes sense, thanks for clarifying!